packages/astro/src/core/app/middlewares.ts

```diff
@@ -25,15 +27,11 @@ export function createOriginCheckMiddleware(): MiddlewareHandler {
 		if (isPrerendered) {
 			return next();
 		}
-		if (request.method === 'GET') {
+		// Safe methods don't require origin check
+		if (SAFE_METHODS.includes(request.method)) {
 			return next();
 		}
-		const sameOrigin =
-			(request.method === 'POST' ||
-				request.method === 'PUT' ||
-				request.method === 'PATCH' ||
-				request.method === 'DELETE') &&
-			request.headers.get('origin') === url.origin;
+		const sameOrigin = request.headers.get('origin') === url.origin;
```

Comment:
a nuance, imho `isSameOrigin` would fit better, as seen also in `isPrerendered`, `hasContentType` etc.